grid2op/tests/test_alert_trust_score.py

```diff
@@ -0,0 +1,1767 @@
+# Copyright (c) 2019-2023, RTE (https://www.rte-france.com)
+# See AUTHORS.txt
+# This Source Code Form is subject to the terms of the Mozilla Public License, version 2.0.
+# If a copy of the Mozilla Public License, version 2.0 was not distributed with this file,
+# you can obtain one at http://mozilla.org/MPL/2.0/.
+# SPDX-License-Identifier: MPL-2.0
+# This file is part of Grid2Op, Grid2Op a testbed platform to model sequential decision making in power systems.
+
+import warnings
+import numpy as np
+import unittest
+import os
+import tempfile
+from grid2op.Observation import BaseObservation
+from grid2op.tests.helper_path_test import *
+
+from grid2op import make
+from grid2op.Reward import _AlertTrustScore
+from grid2op.Parameters import Parameters
+from grid2op.Exceptions import Grid2OpException
+from grid2op.Runner import Runner  # TODO
+from grid2op.Opponent import BaseOpponent, GeometricOpponent, GeometricOpponentMultiArea
+from grid2op.Action import BaseAction, PlayableAction
+from grid2op.Agent import BaseAgent
+from grid2op.Episode import EpisodeData
+
+
+ALL_ATTACKABLE_LINES= [
+            "62_58_180",
+            "62_63_160",
+            "48_50_136",
+            "48_53_141",
+            "41_48_131",
+            "39_41_121",
+            "43_44_125",
+            "44_45_126",
+            "34_35_110",
+            "54_58_154",
+        ] 
+
+ATTACKED_LINE = "48_50_136"
+
+
+#DEFAULT_PARAMS_TRUSTSCORE = dict(reward_min_no_blackout=-1.0,
+#                                 reward_min_blackout=-10.0,
+#                                 reward_max_no_blackout=1.0,
+#                                 reward_max_blackout=2.0,
+#                                 reward_end_episode_bonus=42.0,
+#                                 min_score=-1.0)
+
+DEFAULT_PARAMS_TRUSTSCORE = dict(reward_min_no_blackout=-1.0,
+                                 reward_min_blackout=-50.0,
+                                 reward_max_no_blackout=0.0,
+                                 reward_max_blackout=0.0,
+                                 reward_end_episode_bonus=0.0,
+                                 min_score=-3.0)
+
+def _get_steps_attack(kwargs_opponent, multi=False):
+    """computes the steps for which there will be attacks"""
+    ts_attack = np.array(kwargs_opponent["steps_attack"])
+    res = []
+    for i, ts in enumerate(ts_attack):
+        if not multi:
+            res.append(ts + np.arange(kwargs_opponent["duration"]))
+        else:
+            res.append(ts + np.arange(kwargs_opponent["duration"][i]))
+    return np.unique(np.concatenate(res).flatten())
```

Comment:
no copy paste!

Reply:
from where is the copy paste ? this was like this but I can change

Reply:
thanks for doing it